pkg/router/router_test.go

```diff
@@ -157,24 +163,104 @@ func TestAdmissionEdgeCases(t *testing.T) {
 	}
 }
 
+func TestConfigTemplateExecution(t *testing.T) {
+	// watching for errors
+	caughtErrors := []error{}
+	errCh, stopCh := make(chan error), make(chan struct{})
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func(ch chan error, stop chan struct{}, wg *sync.WaitGroup) {
+		defer wg.Done()
+		for {
+			select {
+			case err := <-ch:
+				caughtErrors = append(caughtErrors, err)
+			case <-stop:
+				return
+			}
+		}
+	}(errCh, stopCh, wg)
+
+	// adding custom handler which pipes to the error channel
+	errHandlersBefore := utilruntime.ErrorHandlers
+	utilruntime.ErrorHandlers = append(utilruntime.ErrorHandlers, (&pipeErrorHandler{errCh}).handle)
+	defer func() { utilruntime.ErrorHandlers = errHandlersBefore }()
+
+	// create routes whose settings would add some additional blocks to the conf
+	start := time.Now()
+	tests := map[string][]expectation{
+		"long whitelist of IPs": {
+			mustCreate{
+				name: "w",
+				host: "anotherexample.com",
+				path: "",
+				time: start,
+				annotations: map[string]string{
+					"haproxy.router.openshift.io/ip_whitelist": getDummyIPs(100),
+				},
+				tlsTermination: routev1.TLSTerminationEdge,
+			},
+		},
+	}
+
+	for name, expectations := range tests {
+		for _, expectation := range expectations {
+			err := expectation.Apply(h)
+			if err != nil {
+				t.Fatalf("%s failed: %v", name, err)
+			}
+		}
+	}
+
+	// let the router reload
+	time.Sleep(reloadInterval * 2)
+
+	stopCh <- struct{}{}
+	wg.Wait()
+
+	// check for errors
+	for _, e := range caughtErrors {
+		if strings.Contains(e.Error(), "error executing template") {
+			t.Fatalf("Template execution failed: %v", e)
+		}
+	}
+
+	if err := cleanUpRoutes(h); err != nil {
+		t.Errorf("Failed to clean up: %v", err)
+	}
```

Comment:
Would it make sense to do this in every test in this file that adds routes?  

This won't be executed if the above code calls `t.Fatalf`, which terminates execution of the test.  Could the call to `cleanupRoutes` be done using `defer` to ensure it always gets executed?

You could simplify usage of `cleanupRoutes` by adding a parameter for `t`, removing `h` (which is already a global), and calling `t.Errorf` inside `cleanupRoutes`, so callers could simply use `defer cleanupRoutes(t)` right after adding a bunch of routes.

Reply:
Done